### ISSUE TITLE
ClusterPointIndicesDecomposer extremely slow for sparse ClusterPoints index

### ISSUE THREAD

**Issue by Affonso-Gui** [jsk-ros-pkg/jsk_recognition#2325]
Hello,

I am using `LabelToClusterPointIndices` -> `ClusterPointIndicesDecomposer` to extract bounding boxes and centroid coordinates from a label image and point cloud.

Since the detection node I am using has many classes, the label values become very sparse (something like three objects being detected with classes `31, 42, 64`), which is making the cluster decomposer extremely slow.

*Some numbers*

When working with `~target` ClusterPointIndices input of about **4~5 Hz**, using original label images results in **0.06 Hz** for `centroid_pose_array`.
With the same input frequency, using labels assigned from 1 in sequence (like `1,2,3,...`) results in about **1.2 Hz**.

*More details*
In `jsk_recognition_msgs/ClusterPointIndices`,  the label number seems to be represented as the index of `cluster_indices` array. Since the labels are sparse, we end up with a very big array fulled with items with no points at all (illustrated below).
```
cluster_indices: 
  - 
    header: 
      seq: 381
      stamp: 
        secs: 1535618326
        nsecs:   4453650
      frame_id: "detection_image"
    indices: []
  - 
    header: settings
      seq: 381
      stamp: 
        secs: 1535618326
        nsecs:   4453650
      frame_id: "detection_image"
    indices: []
  - 105043, 105044, 105045, 105046,
(...)
  - 
    header: 
      seq: 381
      stamp: 
        secs: 1535618326
        nsecs:   4453650
      frame_id: "detection_image"
    indices: [105036, 105037, 105038, 105039, 105040, 105041, 105042,  (...)
```
Since null items should not actually demand any calculation, the speed is expected to be somewhat the same to both sparse and dense labeling. However, that does not seems to be the case. 

 Maybe spending too much time in sorting or overall setup? 
Setting `min_size` did not show any relevant differences.

**Comment by iory** [comment]
This seems to be occurring in the function that outputs negative_indices.
I created PR here https://github.com/jsk-ros-pkg/jsk_recognition/pull/2326

This PR adds publish_negative_indices param to cluster_point_indices.

Please try this PR and review it.

**Comment by Affonso-Gui** [comment]
Thanks! I will test this tomorrow.

**Comment by Affonso-Gui** [comment]
Closed via #2326
